firewood/src/hashednode.rs

```diff
@@ -22,7 +23,9 @@ use storage::{ReadLinearStore, WriteLinearStore};
 #[derive(Debug)]
 pub struct HashedNodeStore<T: ReadLinearStore> {
     nodestore: NodeStore<T>,
-    modified: HashMap<LinearAddress, (Arc<Node>, u8)>,
+    // Linear Address -> (Node, Node Size Index)
+    // None means the node was deleted.
+    modified: HashMap<LinearAddress, Option<(Arc<Node>, u8)>>,
```

Comment:
I think this is not needed. Why can't you just remove it from modified when it is deleted?

Nobody should ever reference it once it has been deleted in any case.

Reply:
What does the `delete_node` implementation look like without this? 

> Why can't you just remove it from modified when it is deleted?
What if the node being deleted wasn't modified in this `HashedNodeStore`? Then removing it from `modified` would be a no-op.

If the node _is_ in `modified`, and we delete it, what happens then? When do we tell the underlying node store to delete the node at the given address?

Reply:
> If the node _is_ in `modified`, and we delete it, what happens then? When do we tell the underlying node store to delete the node at the given address?

Yes. We should just delete it from the linear store.

Reply:
>  If you don't do this, then the space for it can't be reused by another node in the same batch, for example.

Yeah that's definitely a bug in this PR. Going to close this. I removed these changes from #644